<!-- Launch file for the rrt-detector and the assigner -->


<launch>
    <arg name="eta" value="1.0"/>
    <arg name="Geta" value="1.0"/>
    <arg name="info_radius" value="1.0"/>
    <arg name="info_multiplier" value="5.0"/> <!--3.0-->
    <arg name="hysteresis_radius" value="10.0"/>   <!--3.0, 5.0-->
    <arg name="hysteresis_gain" value="5.0"/>   <!--3.0, 5.0-->
    <arg name="costmap_clearing_threshold" value="50"/>
    <arg name="delay_after_assignement" value="0.7"/>   
    <arg name="bandwith_cluster" value="1.0"/>   <!--1.0-->
    <arg name="frequency1" value="5.0"/>
    <arg name="detected_frontier" value="/detected_points"/>
    <arg name="filtered_frontier" value="/filtered_points"/>
    <!-- <arg name="rd_metric" value="15.0"/> -->
    <!-- robot name space definition -->
    <arg name="first_tb3"  default=""/>
    <arg name="raw_global_map" value ="/projected_map"/>
    <arg name="erode_kernel_size" value ="5"/>  <!--5.0 for 17W, 7.0 for 1W-->
    <arg name="global_map" value ="/projected_map_erode"/>
    <arg name="global_frame" value ="map"/>
    <arg name="global_costmap_topic" value="/move_base_node/global_costmap/costmap"/>
    <arg name="frontiers_topic" value ="/filtered_points"/>
    <arg name="goal_topic" value ="/nbv_points"/>
    <arg name="finished_topic" value ="/finished_goal"/>
    <arg name="inter_points_topic" value ="/inter_points"/>
    <arg name="viewpoint_topic" value ="/viewpoints"/>
    <arg name="viewpoint_finished_topic" value ="/finished_viewpoint"/>
    <arg name="planner_topic" value ="/way_point"/>
    <arg name="distance_map_type" value ="distmap/DistanceMapDeadReck"/>
    <arg name="dist_map_topic" value ="/distance_map_node/distance_field_obstacles"/>
    <arg name="query_map_topic" value ="/query_map"/>
    
    <arg name="model" default="waffle_pi" doc="model type [burger, waffle, waffle_pi]"/>
    <arg name="move_forward_only" default="false"/>
    <!-- <arg name="xMax" default="10.0"/>
    <arg name="xMin" default="10.0"/>
    <arg name="yMax" default="10.0"/>
    <arg name="yMin" default="10.0"/> -->
    <arg name="xRange" default="10.0"/>
    <arg name="yRange" default="10.0"/>
    <arg name="curX" default="0.0"/>
    <arg name="curY" default="0.0"/>

    <!--
    <node pkg="rrt_exploration" type="my_tf" name="my_tf" output="screen">
    </node>
    -->
    
    <node pkg="rrt_exploration" type="map_erode.py" name="map_erode" output="screen">
      <param name="raw_map_topic" value="$(arg raw_global_map)"/>
      <param name="map_topic" value="$(arg global_map)"/>
      <param name="erode_kernel_size" value="$(arg erode_kernel_size)"/>
    </node>

    
    <node pkg="distance_map_node" type="distance_map_node" name="distance_map_node" output="screen">
      <param name="distance_map_type" value="$(arg distance_map_type)" />
      <remap from="/map" to="$(arg query_map_topic)" />
    </node>
    

    <node pkg="rrt_exploration" type="global_rrt_detector" name="global_detector" output="screen">
      <param name="eta" value="$(arg Geta)"/>
      <param name="map_topic" value="$(arg global_map)"/>
      <param name="x_range" value="$(arg xRange)"/>
      <param name="y_range" value="$(arg yRange)"/>
      <param name="cur_x" value="$(arg curX)"/>
      <param name="cur_y" value="$(arg curY)"/>
      <param name="inter_points_topic" value ="$(arg inter_points_topic)"/>
    </node>

    
    <node pkg="rrt_exploration" type="local_rrt_detector" name="local_detector" output="screen">
      <param name="eta" value="$(arg eta)"/>
      <param name="map_topic" value="$(arg global_map)"/>
      <param name="robot_frame" value="base_link"/>
      <param name="x_range" value="$(arg xRange)"/>
      <param name="y_range" value="$(arg yRange)"/>
      <param name="cur_x" value="$(arg curX)"/>
      <param name="cur_y" value="$(arg curY)"/>
    </node>
 
    
    <!-- for the filter node  -->
    
    <node pkg="rrt_exploration" type="filter.py" name="filter" output="screen">
      <param name="map_topic" value="$(arg global_map)"/>
      <param name="info_radius" value="$(arg info_radius)"/> 
      <param name="costmap_clearing_threshold" value="$(arg costmap_clearing_threshold)"/> 
      <param name="global_costmap_topic" value="$(arg global_costmap_topic)"/> 
      <param name="goals_topic" value="$(arg detected_frontier)"/> 
      <param name="robot_namelist" value="$(arg first_tb3)"/>
      <param name="rate" value="$(arg frequency1)"/>
      <param name="local_map" value="map"/>
      <param name="bandwith_cluster" value="$(arg bandwith_cluster)"/>
    </node>

    
    <node pkg="rrt_exploration" type="nbvp.py" name="nbvp" output="screen">
      <param name="info_radius" value="$(arg info_radius)"/> 
      <param name="info_multiplier" value="$(arg info_multiplier)"/> 
      <param name="hysteresis_radius" value="$(arg hysteresis_radius)"/> 
      <param name="hysteresis_gain" value="$(arg hysteresis_gain)"/> 
      <param name="frontiers_topic" value="/filtered_points"/> 
      <param name="delay_after_assignement" value="$(arg delay_after_assignement)"/>

      <param name="map_topic" value="$(arg global_map)"/> 
      <param name="frontiers_topic" value="$(arg frontiers_topic)"/> 
      <param name="goal_topic" value="$(arg goal_topic)"/> 
      <param name="finished_topic" value="$(arg finished_topic)"/> 
      <param name="inter_points_topic" value="$(arg inter_points_topic)"/> 
    </node>

    
    <node pkg="rrt_exploration" type="local_planner.py" name="local_planner" output="screen">
      <param name="map_topic" value="$(arg global_map)"/> 
      <param name="viewpoint_finished_topic" value="$(arg viewpoint_finished_topic)"/> 
      <param name="viewpoint_topic" value="$(arg viewpoint_topic)"/> 
      <param name="goal_topic" value="$(arg goal_topic)"/> 
      <param name="finished_topic" value="$(arg finished_topic)"/> 
      <param name="planner_topic" value="$(arg planner_topic)"/> 
      <param name="dist_map_topic" value="$(arg dist_map_topic)"/> 
    </node>
    
    
</launch>
